controlplane/internal/imageregistry/image_registry.go

```diff
@@ -112,10 +113,15 @@ func getImageRegistries(registry string) (string, string, []string, error) {
 	for _, registry := range registriesTree {
 		source, sourceExists := registry.Get("location").(string)
 		mirrorTrees, mirrorExists := registry.Get("mirror").([]*toml.Tree)
-		if !sourceExists || !mirrorExists {
-			continue
+		if !sourceExists {
+			return "", "", nil, fmt.Errorf("failed to parse image registry toml, missing registry location")
+		}
+		if !mirrorExists {
+			return "", "", nil, fmt.Errorf("failed to parse image registry toml, missing registry.mirror in registries.conf")
+		}
+		if err := parseMirrorRegistries(&idmsMirrors, &itmsMirrors, mirrorTrees, &insecureRegistries, source); err != nil {
+			return "", "", nil, errors.Wrap(err, fmt.Sprintf("failed to parse image mirror registry toml for registry %s", source))
 		}
```

Comment:
I think we can reorganize the `getImageRegistries` function, maybe using a specific struct to encapsulate values from current `parse` function (that will need renaming). For example:
```golang
type RegistryConfig struct {
    TagMirrors *configv1.ImageTagMirrors{}
    DigestMirrors *configv1.ImageDigestMirrors{}
    InsecureRegistries []string
}
...

configs := []RegistryConfig{}
for _, registry := range registriesTree {
	source, sourceExists := registry.Get("location").(string)
	mirrorTrees, mirrorExists := registry.Get("mirror").([]*toml.Tree)
        // handle errors
        if registryConfig, err := getRegistryConfig(source, mirrorTrees); err != nil {
        // handle errors
        }
        configs = append(configs, registryConfig)
}

...
generateImageDigestMirrorSet(configs)
generateImageTagMirrorSet(configs)
generateInsecureRegistries(configs)
generateOpenshiftImageConfig(configs, registryCertExists)
```

Or something along these lines. WDYT?

Reply:
Good idea! Modified it, how does it look?